lib/govuk_publishing_components/presenters/public_layout_helper.rb

```diff
@@ -69,20 +100,43 @@ def footer_track_actions
         FOOTER_TRACK_ACTIONS
       end
 
-      def generate_data_attribute(link, track_action)
+      def generate_data_attribute(link, track_action, ga4_attributes)
+        if ga4_attributes
+          ga4_link = {
+            "event_name": "navigation",
+            "type": "footer",
+            "index": {
+              "index_link": ga4_attributes[:index_link].to_s,
+              "index_section": ga4_attributes[:index_section].to_s,
+              "index_section_count": "5",
+            }, # //Section.link number notation
+            "index-total": @footer_links_count.to_s, # //Across all sections
+            "section": ga4_attributes[:section],
+            # //Use 'Support links', 'Licence' and 'Copyright' for sections 3, 4 & 5
+          }
+        end
+
         {
           track_category: "footerClicked",
           track_action: track_action,
           track_label: link[:href],
           track_options: {
             dimension29: link[:text],
           },
+          ga4_link: ga4_link,
         }
       end
 
-      def add_data_attributes_to_links(items, track_action)
+      def add_data_attributes_to_links(items, track_action, ga4_data)
+        ga4_index = 0
         items.map do |item|
```

Comment:
Quick question -  would using `items.map.with_index do |item, index|` be an option here? I did some quick local testing and it seemed to work fine but no worries if there's a reason why we're not using it - just wanted to check.

Reply:
@JamesCGDS Good spot - I tried `map_with_index` but didn't realise it was actually `map.with_index` so I'll update it to use that :+1: